File: Data Structures/Stack.java
public class Stack<T> {
    private Node<T> head;
    private int count;
    public Stack() {
        this.head = null;
        this.count = 0;
    }
    public Stack(T data) {
        this.head = new Node(data);
        this.count = 1;
    }
    public boolean push(T data) {
        if (head == null) {
            head = new Node(data);
            return true;
        }
        Node temp = new Node(data);
        temp.next = head;
        head = temp;
    }
    public T pop() {
        if (head == null) {
            return null;
        }
        Node temp = head;
        head = temp.next;
        return temp.get();
    }
    public T peek() {
        if (head == null)
            return null;
        return head.get();
    }
    public int size() {
        return count;
    }
}

File: Python/Data_Structures/Stack/Stack.py
from Data_Structures.Node.Node import *

class Stack():
    def __init__(self):
        self.head = None
        self.count = 0

    def push(self, value):
        if (self.head == None):
            self.head = Node(value)
        else:
            temp = Node(value)
            temp.next = self.head
            self.head = temp
        self.count = self.count + 1
        return True

    def peek(self):
        if (self.head == None):
            return None
        else:
            return self.head.val

    def pop(self):
        if (self.head == None):
            return None
        else:
            temp = self.head.val
            self.head = self.head.next
            self.count = self.count - 1
            return temp

    def isEmpty(self):
        return (self.count > 0)

    def clear(self):
        self.head = None
        self.count = 0
